### PR TITLE
Fix rds db param group default

### PR DESCRIPTION
**Pull request by VenelinMartinov**
In https://github.com/pulumi/pulumi-aws/pull/3638 - we introduced a diff customizer to change the schema as an alternative to patching. The change does not pass InternalValidate but because of https://github.com/pulumi/pulumi-terraform-bridge/issues/1852 it was not flagged up.

After turning InternalValidate back on (https://github.com/pulumi/pulumi-terraform-bridge/issues/1852) we discovered an issue with this parameter:

```
make tfgen_no_deps
(cd provider && go build -p 2 -o /Users/vvm/code/pulumi-aws/bin/pulumi-tfgen-aws -ldflags "-X github.com/pulumi/pulumi-aws/provider/v6/pkg/version.Version=6.29.1+dirty" github.com/pulumi/pulumi-aws/provider/v6/cmd/pulumi-tfgen-aws)
/Users/vvm/code/pulumi-aws/bin/pulumi-tfgen-aws schema --out provider/cmd/pulumi-resource-aws
Errors occurred: [InternalValidate: Internal validation of the provider failed! This is always a bug
with the provider itself, and not a user issue. Please report
this bug:

resource aws_db_parameter_group: apply_method: Default must be nil if computed]
make: *** [tfgen_no_deps] Error 255
```

This should fix it by also setting the Default to nil in the schema.

To keep the current behaviour, we set the Default in the pulumi overlay instead.

### CODE REVIEW

File: sdk/dotnet/Rds/Inputs/ParameterGroupParameterArgs.cs [hunk 1]
```diff
@@ -34,6 +34,7 @@ public sealed class ParameterGroupParameterArgs : global::Pulumi.ResourceArgs
 
         public ParameterGroupParameterArgs()
         {
+            ApplyMethod = "immediate";
```
Comment (VenelinMartinov) [inline sdk/dotnet/Rds/Inputs/ParameterGroupParameterArgs.cs:37]: I wonder why we didn't generate the defaults in the sdk before - this PR just moves the defaults from the TF schema to the pulumi overlay. IMO this is a better UX for the sdks, does anyone know why they didn't show up before?

Comment (iwahbe) [reply sdk/dotnet/Rds/Inputs/ParameterGroupParameterArgs.cs:37]: I believe it depends on how the defaults are applied. We can't always see inside TF providers to find the defaults they supply.

**Review by iwahbe — COMMENTED** [review]
I'm having trouble understanding this PR. 

Is this a bug we introduced in AWS in a patch, and if so can we just fix the patch? If not, does this imply that the upstream schema is invalid, and if so can we submit an upstream patch instead?

**Comment by VenelinMartinov** [comment]
The context is this: https://github.com/pulumi/pulumi-aws/pull/3638 - we introduced a diff customizer to change the schema as an alternative to patching. The change does not pass InternalValidate but because of https://github.com/pulumi/pulumi-terraform-bridge/issues/1852 it was not flagged up.

This PR corrects the change to make the schema pass InternalValidate

**Review by iwahbe — APPROVED** [review]

File: provider/provider_yaml_test.go [hunk 2]
```diff
@@ -184,13 +168,67 @@ resources:
...
+  randSuffix:
+    type: string
+resources:
+  default:
+    type: aws:rds/parameterGroup:ParameterGroup
+    properties:
+      name: securitygroup${randSuffix}
+      family: postgres14
+      parameters:
+        - name: track_io_timing
```
Comment (t0yv0) [inline provider/provider_yaml_test.go:186]: Nice this looks like it's testing the case where the default might kick in.

File: provider/resources.go [hunk 3]
```diff
@@ -3093,6 +3093,18 @@ func ProviderFromMeta(metaInfo *tfbridge.MetadataInfo) *tfbridge.ProviderInfo {
...
 						Default: managedByPulumi,
 					},
+					"parameter" : {
+						Elem: &tfbridge.SchemaInfo{
+							Fields: map[string]*tfbridge.SchemaInfo{
+								"apply_method": {
+									// We set the default value in the overlay since
+									// we remove it in the TF schema in 
+									// provider/pkg/rds/parameter_group.go
+									Default: &tfbridge.DefaultInfo{Value: "immediate"},
```
Comment (t0yv0) [inline provider/resources.go:3103]: Looks like this compensates for the patch. Very nice.

**Review by t0yv0 — APPROVED** [review]
🚢

**Comment by t0yv0** [comment]
Yes @iwahbe the alternative here would be to rework https://github.com/pulumi/pulumi-aws/issues/2442 fix with patching, instead of a diff customizer, but as things stand this change here is the lightest-way step forward, so I think this is good for now.

**Pull request merged**